tests/serverpod_test_server/test_e2e/service_protocol_test.dart

```diff
@@ -465,6 +465,23 @@ void main() {
       });
     });
   });
+
+  group('File retrieval', () {
+    test('Fetch generated/protocol.yaml file', () async {
+      var file =
+          await serviceClient.insights.fetchFile('generated/protocol.yaml');
+      expect(file, isNotNull);
+      expect(file.length, greaterThan(0));
+    });
+    test('Fetch file outside whitelist', () async {
+      expect(
+        () async {
+          await serviceClient.insights.fetchFile('non-existant-file');
```

Comment:
Suggestion: Use a file we know exists but that we don't want to expose.

The main scenario we want to assert here is that we don't allow access to files that should not be accessible. I think it would be wise to use a file we know that we don't want to expose to catch that issue.

Reply:
Smart, fixing this!